### PR TITLE
grt: fix loops on local connections for repair_antennas

### PR DESCRIPTION
**Pull request by eder-matheus**
Fix https://github.com/The-OpenROAD-Project/OpenLane/issues/929

### CODE REVIEW

File: src/grt/src/GlobalRouter.tcl [hunk 1]
```diff
@@ -144,6 +144,18 @@ proc set_macro_extension { args } {
   }
 }
 
+sta::define_cmd_args "set_pin_offset" { offset }
```
Comment (maliberty) [inline src/grt/src/GlobalRouter.tcl:147]: What is the purpose of this command?  Does it need to be documented or is just for developer use?

File: src/grt/src/GlobalRouter.cpp [hunk 2]
```diff
@@ -2040,6 +2042,52 @@ void GlobalRouter::addLocalConnections(NetRouteMap& routes)
...
 }
 
+bool GlobalRouter::pinOverlapsGSegment(const odb::Point& pin_position,
+                                       const int pin_layer,
+                                       const std::vector<odb::Rect>& pin_boxes,
+                                       const GRoute& route)
+{
+  bool segment_overlaps_pin = false;
+
+  // check if pin position on grid overlaps with the pin shape
```
Comment (maliberty) [inline src/grt/src/GlobalRouter.cpp:2052]: Are these two checks supposed to be both required or either required?  segment_overlaps_pin  may get overwritten in the 2nd check.

File: src/grt/src/GlobalRouter.cpp [hunk 3]
```diff
@@ -2040,6 +2042,52 @@ void GlobalRouter::addLocalConnections(NetRouteMap& routes)
...
+        int x0 = std::min(seg.init_x, seg.final_x);
+        int y0 = std::min(seg.init_y, seg.final_y);
+        int x1 = std::max(seg.init_x, seg.final_x);
+        int y1 = std::max(seg.init_y, seg.final_y);
+
+        if (x0 == x1) { // vertical segment
+          if ((segment_overlaps_pin = box.xMin() <= x0 && box.xMax() >= x0 &&
+                                 ((box.yMin() <= y0 && box.yMax() >= y0) || 
+                                  (box.yMin() <= y1 && box.yMax() >= y1) ||
+                                  (box.yMax() >= y0 && box.yMin() <= y1)))) {
```
Comment (maliberty) [inline src/grt/src/GlobalRouter.cpp:2073]: Just make a 2nd Rect and use contains which will simplify this code.

File: src/grt/src/GlobalRouter.tcl (hunk 1 above)
Comment (eder-matheus) [reply src/grt/src/GlobalRouter.tcl:147]: It is just for developer use. It defines the offset (in gcells) that the fake pins will have from the original pin position.

**Review by maliberty — APPROVED** [review]

**Pull request merged**